File: packages/health_connector_hc_android/android/src/main/kotlin/com/phamtunglam/health_connector_hc_android/mappers/health_measurement_unit_mappers/BloodGlucoseMapper.kt
package com.phamtunglam.health_connector_hc_android.mappers.health_measurement_unit_mappers

import androidx.health.connect.client.units.BloodGlucose
import com.phamtunglam.health_connector_hc_android.pigeon.BloodGlucoseDto

/**
 * Converts a [BloodGlucoseDto] to a Health Connect [BloodGlucose] object.
 */
internal fun BloodGlucoseDto.toHealthConnect(): BloodGlucose =
    BloodGlucose.millimolesPerLiter(millimolesPerLiter)

/**
 * Converts a Health Connect [BloodGlucose] object to a [BloodGlucoseDto].
 *
 * Uses millimoles per liter as the transfer unit for consistency.
 */
internal fun BloodGlucose.toDto(): BloodGlucoseDto = BloodGlucoseDto(
    millimolesPerLiter = inMillimolesPerLiter,
)

File: packages/health_connector_hc_android/lib/src/mappers/measurement_unit_mappers/blood_glucose_mapper.dart
import 'package:health_connector_core/health_connector_core_internal.dart'
    show BloodGlucose;
import 'package:health_connector_hc_android/src/pigeon/health_connector_hc_android_api.g.dart'
    show BloodGlucoseDto;
import 'package:meta/meta.dart' show internal;

/// Converts [BloodGlucose] to [BloodGlucoseDto].
@internal
extension BloodGlucoseToDto on BloodGlucose {
  BloodGlucoseDto toDto() {
    // Uses millimoles per liter as the transfer unit for consistency.
    return BloodGlucoseDto(
      millimolesPerLiter: inMillimolesPerLiter,
    );
  }
}

/// Converts [BloodGlucoseDto] to [BloodGlucose].
@internal
extension BloodGlucoseDtoToDomain on BloodGlucoseDto {
  BloodGlucose toDomain() {
    return BloodGlucose.millimolesPerLiter(millimolesPerLiter);
  }
}
